presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -676,11 +816,45 @@ public void abort()
         return future;
     }
 
+    public static StreamingPlanSection extractStreamingSections(SubPlan subPlan)
+    {
+        List<SubPlan> childrenRoots = new ArrayList<>();
+        StreamingSubPlan streamingSection = extractStreamingSection(subPlan, childrenRoots);
+        return new StreamingPlanSection(
+                streamingSection,
+                childrenRoots.stream()
+                        .map(SqlQueryScheduler::extractStreamingSections)
+                        .collect(toImmutableList()));
+    }
+
+    private static StreamingSubPlan extractStreamingSection(SubPlan subPlan, List<SubPlan> otherRoots)
```

Comment:
What do you think about making otherRoots an `ImmutableList.Builder<SubPlan>`, it might make it clear it's more like an out parameter.

Reply:
@tdcmeehan : Good point ! This was did for `createStages` (e.g. `stageSchedulers`, `stageLinkages` ) :)